File: sdk-go/models/vault.go
package models

// Vault represents a trading vault.
type Vault struct {
	Address          string    `json:"address"`
	Name             string    `json:"name"`
	Description      *string   `json:"description"`
	Manager          string    `json:"manager"`
	Status           string    `json:"status"`
	CreatedAt        int64     `json:"created_at"`
	TVL              *float64  `json:"tvl"`
	Volume           *float64  `json:"volume"`
	Volume30d        *float64  `json:"volume_30d"`
	AllTimePnl       *float64  `json:"all_time_pnl"`
	NetDeposits      *float64  `json:"net_deposits"`
	AllTimeReturn    *float64  `json:"all_time_return"`
	PastMonthReturn  *float64  `json:"past_month_return"`
	SharpeRatio      *float64  `json:"sharpe_ratio"`
	MaxDrawdown      *float64  `json:"max_drawdown"`
	WeeklyWinRate    *float64  `json:"weekly_win_rate_12w"`
	ProfitShare      *float64  `json:"profit_share"`
	Pnl90d           *float64  `json:"pnl_90d"`
	ManagerCashPct   *float64  `json:"manager_cash_pct"`
	AverageLeverage  *float64  `json:"average_leverage"`
	Depositors       *int64    `json:"depositors"`
	PerpEquity       *float64  `json:"perp_equity"`
	VaultType        *string   `json:"vault_type"`
	SocialLinks      []string  `json:"social_links"`
}

// VaultsResponse is the response for listing vaults.
type VaultsResponse struct {
	Items            []Vault `json:"items"`
	TotalCount       int64   `json:"total_count"`
	TotalValueLocked float64 `json:"total_value_locked"`
	TotalVolume      float64 `json:"total_volume"`
}

// UserOwnedVault represents a vault owned by a user.
type UserOwnedVault struct {
	VaultAddress     string   `json:"vault_address"`
	VaultName        string   `json:"vault_name"`
	VaultShareSymbol string   `json:"vault_share_symbol"`
	Status           string   `json:"status"`
	AgeDays          int64    `json:"age_days"`
	NumManagers      int64    `json:"num_managers"`
	TVL              *float64 `json:"tvl"`
	APR              *float64 `json:"apr"`
	ManagerEquity    *float64 `json:"manager_equity"`
	ManagerStake     *float64 `json:"manager_stake"`
}

File: decibel-sdk-python/decibel/models/vault.py
"""Vault-related data models."""

from typing import Any

from pydantic import BaseModel, Field

from .enums import VaultType


class Vault(BaseModel):
    """Trading vault.

    Attributes:
        address: Vault address
        name: Vault name
        description: Vault description
        manager: Manager account address
        status: Vault status
        created_at: Creation timestamp
        tvl: Total value locked
        volume: Trading volume
        volume_30d: 30-day volume
        all_time_pnl: All-time PnL
        net_deposits: Net deposits
        all_time_return: All-time return
        past_month_return: Past month return
        sharpe_ratio: Sharpe ratio
        max_drawdown: Maximum drawdown
        weekly_win_rate_12w: 12-week weekly win rate
        profit_share: Profit share percentage
        pnl_90d: 90-day PnL
        manager_cash_pct: Manager cash percentage
        average_leverage: Average leverage
        depositors: Number of depositors
        perp_equity: Perp equity
        vault_type: Vault type
        social_links: Social media links
    """

    address: str
    name: str
    manager: str
    status: str
    created_at: int
    description: str | None = None
    tvl: float | None = None
    volume: float | None = None
    volume_30d: float | None = None
    all_time_pnl: float | None = None
    net_deposits: float | None = None
    all_time_return: float | None = None
    past_month_return: float | None = None
    sharpe_ratio: float | None = None
    max_drawdown: float | None = None
    weekly_win_rate_12w: float | None = None
    profit_share: float | None = None
    pnl_90d: float | None = None
    manager_cash_pct: float | None = None
    average_leverage: float | None = None
    depositors: int | None = None
    perp_equity: float | None = None
    vault_type: VaultType | None = None
    social_links: list[str] = Field(default_factory=list)


class VaultsResponse(BaseModel):
    """Response for listing vaults.

    Attributes:
        items: List of vaults
        total_count: Total count
        total_value_locked: Total value locked across all vaults
        total_volume: Total volume across all vaults
    """

    items: list[Vault]
    total_count: int
    total_value_locked: float
    total_volume: float


class UserOwnedVault(BaseModel):
    """Vault owned by a user.

    Attributes:
        vault_address: Vault address
        vault_name: Vault name
        vault_share_symbol: Share token symbol
        status: Status
        age_days: Age in days
        num_managers: Number of managers
        tvl: Total value locked
        apr: Annual percentage rate
        manager_equity: Manager equity
        manager_stake: Manager stake
    """

    vault_address: str
    vault_name: str
    vault_share_symbol: str
    status: str
    age_days: int
    num_managers: int
    tvl: float | None = None
    apr: float | None = None
    manager_equity: float | None = None
    manager_stake: float | None = None


class UserPerformanceOnVault(BaseModel):
    """User's performance on a vault.

    Attributes:
        vault_address: Vault address
        vault_name: Vault name
        shares: Number of shares owned
        share_value: Value per share
        total_value: Total value of shares
        deposited_amount: Total deposited amount
        withdrawn_amount: Total withdrawn amount
        realized_pnl: Realized PnL
        unrealized_pnl: Unrealized PnL
        all_time_return: All-time return
    """

    vault_address: str
    vault_name: str
    shares: int
    share_value: float
    total_value: float
    deposited_amount: float
    withdrawn_amount: float
    realized_pnl: float
    unrealized_pnl: float
    all_time_return: float
